### PR TITLE
fix: check deployed status of a plugin before creating the configuration

### PR DESCRIPTION
**Pull request by leleueri**
fixes AM-732

## description

This PR aims to reject a EE plugin creation if the plugin is not deployed (plugin not part of the license or missing from the installation)
Check apply on:
* IdentityProvider
* BotDetection
* Factor
* Resources
* Policies
* Reporter
* ExtensionGrantType

## How to test

On EE environment
* Open network view of the brower
* create EE IDP
* copy the curl command from the network view
* do the same for EE factor, EE resource, and a EE plugin.

On CE environmen:
* use the curl commands to have the right payloas (adapt the internal ids)
* the request should fail with the message '`Plugin type xxx not deployed`'

### CODE REVIEW

File: gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/organizations/environments/domains/BotDetectionsResource.java [hunk 1]
```diff
@@ -52,6 +53,9 @@ public class BotDetectionsResource extends AbstractResource {
     @Autowired
     private BotDetectionServiceProxy botDetectionService;
 
+    @Autowired
+    private BotDetectionPluginService botDetectionPluginService;
```
Comment (passionne) [inline gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/organizations/environments/domains/BotDetectionsResource.java:57]: It is not used ? Did I miss somethings ?

Comment (leleueri) [reply gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/organizations/environments/domains/BotDetectionsResource.java:57]: indeed, I forgot to use it in the create method

File: gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/IdentityProviderServiceProxyImpl.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
 import io.gravitee.am.model.ReferenceType;
 import io.gravitee.am.service.AuditService;
 import io.gravitee.am.service.exception.IdentityProviderNotFoundException;
+import io.gravitee.am.service.exception.PluginNotDeployedException;
```
Comment (passionne) [inline gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/IdentityProviderServiceProxyImpl.java:32]:
```suggestion
```
It is not used 😊

Comment (mariano-fernandez) [inline gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/IdentityProviderServiceProxyImpl.java:32]: is this an unused import?

File: gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/plugins/BotDetectionPluginServiceImpl.java [hunk 3]
```diff
@@ -100,4 +103,12 @@ private BotDetectionPlugin convert(Plugin plugin) {
         botDetectionPlugin.setFeature(plugin.manifest().feature());
         return botDetectionPlugin;
     }
+
+    public Completable checkPluginDeployment(String type) {
```
Comment (mariano-fernandez) [inline gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/plugins/BotDetectionPluginServiceImpl.java:107]: If I'm not mistaken, this method is doing exactly the same that is done by the extended class

[the remark this answers is not in the document]
Comment (leleueri) [reply gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/plugins/BotDetectionPluginServiceImpl.java:107]: PluginService created, all XXXPluginService inherit from this new interface.

**Review by mariano-fernandez — APPROVED** [review]

**Pull request merged**